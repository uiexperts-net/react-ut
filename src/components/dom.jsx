import React, { useState } from "react"

export const DOMTesting = () => {
    const [isConnected, setConnect] = useState('NO');

    return (
        <div data-testid='root'>
            <h1> DOM Testing</h1>
            <div>Connect Status : <span data-testid='connect-status'>Status:{isConnected}</span></div>
            <button onClick={()=>setConnect('YES')}>Connect</button>
        </div>
    )

}