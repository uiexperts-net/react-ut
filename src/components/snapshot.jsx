import React from 'react';

export const SnapshotComp = ()=>{
    return(
        <div>
            <h1>Hello from Snapshot</h1>
        </div>
    )
}