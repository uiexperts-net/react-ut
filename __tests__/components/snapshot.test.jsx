import React from 'react';
import renderer from 'react-test-renderer';
import { SnapshotComp } from '../../src/components/snapshot';

describe('snapshot testing to validate changes in file', () => {

    test('validate Snapshot for a component', () => {
        const reactTree = renderer.create(<SnapshotComp />).toJSON();
        expect(reactTree).toMatchSnapshot();
    })

    test('validate InlineSnapshot for a component', () => {
        const reactTree = renderer.create(<SnapshotComp />).toJSON();
        expect(reactTree).toMatchInlineSnapshot(`
<div>
  <h1>
    Hello from Snapshot
  </h1>
</div>
`);

    });

    test('validate Property Matchers for dynamically changing values', () => {
        const user = {
            createdAt: new Date(),
            id: Math.floor(Math.random() * 20),
            name: 'LeBron James',
        };

        expect(user).toMatchSnapshot({
            createdAt: expect.any(Date),
            id: expect.any(Number),
        });
    });
})





