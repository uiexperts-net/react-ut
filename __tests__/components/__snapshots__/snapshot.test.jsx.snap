// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing to validate changes in file validate Property Matchers for dynamically changing values 1`] = `
{
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "LeBron James",
}
`;

exports[`snapshot testing to validate changes in file validate Snapshot for a component 1`] = `
<div>
  <h1>
    Hello from Snapshot
  </h1>
</div>
`;

exports[`unit testing for demo To check react base component 1`] = `
<div>
  <h1>
    Hello from Snapshot
  </h1>
</div>
`;

exports[`unit testing for demo validate Property Matchers for dynamically changing values 1`] = `
{
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "LeBron James",
}
`;

exports[`unit testing for demo validate Snapshot for a component 1`] = `
<div>
  <h1>
    Hello from Snapshot
  </h1>
</div>
`;
